llvm/include/llvm/ProfileData/MemProfReader.h

```diff
@@ -57,6 +57,36 @@ class MemProfReader {
     return FunctionProfileData;
   }
 
+  // Convert IndexedMemProfRecord to MemProfRecord, populating call stacks and
+  // frames inline.
+  virtual MemProfRecord convertIndexedMemProfRecordToMemProfRecord(
```

Comment:
How about making this function a method of the IndexedMemProfRecord class (e.g. `toMemProfRecord`) with only the `Callback` parameter?

Reply:
Done.  Now the call back converts `CallStackId` to `llvm::SmallVector<Frame>`.  That is, the conversion from `FrameId` to `Frame` is handled behind the scenes.